### PR TITLE
Add support to validate endpoints.yaml & component-config.yaml

### PR DESCRIPTION
**Pull request by binoyPeries**
$Subject
Related issues: 

- https://github.com/wso2-enterprise/choreo/issues/30473
- https://github.com/wso2-enterprise/choreo/issues/30531

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,58 +2,101 @@ const core = require("@actions/core");
...
+      case SourceConfigFileTypes.COMPONENT_CONFIG_YAML:
+        fullPath = path.join(
+          fullPath,
+          SourceConfigFileTypes.COMPONENT_CONFIG_YAML
+        );
+        break;
+      case SourceConfigFileTypes.ENDPOINT_YAML:
+        fullPath = path.join(fullPath, SourceConfigFileTypes.ENDPOINT_YAML);
+        break;
+      default:
```
Comment (JanakaSandaruwan) [inline index.js]:
```suggestion
   if (fileType === SourceConfigFileTypes.COMPONENT_CONFIG_YAML || ...) {
fullPath = path.join(
          fullPath,
          fileType
        );
   }
```

File: dist/index.js [hunk 2]
```diff
@@ -38166,58 +38226,101 @@ const core = __nccwpck_require__(2153);
...
+        fullPath = path.join(fullPath, SourceConfigFileTypes.ENDPOINT_YAML);
+        break;
+      default:
+        throw new Error(`'${fileType}' is not a valid source config file type`);
+    }
     let fileContent = fs.readFileSync(fullPath, "utf8");
     return fileContent;
   } catch (error) {
-    throw new Error(`Failed to read component.yaml: ${error.message}`);
+    throw new Error(`\nFailed to read source config file: ${error.message}`);
```
Comment (JanakaSandaruwan) [inline dist/index.js:38269]: Why we add \n here?

File: index.js [hunk 3]
```diff
@@ -2,58 +2,101 @@ const core = require("@actions/core");
...
     // Parse the yaml content
-    componentYamlFile = yaml.load(fileContent);
-    await validateComponentYaml(sourceRootDir);
+    srcConfigYamlFile = yaml.load(fileContent);
+    await validateSourceConfigFile(sourceRootDir, fileType);
   } catch (error) {
-    console.log("component.yaml validation failed: ", error.message);
-    core.setFailed("component.yaml validation failed ", error.message);
+    console.log("Validation Error: ", error.message);
+    core.setFailed("source config file validation failed ", error.message);
```
Comment (JanakaSandaruwan) [inline index.js]:
```suggestion
    core.setFailed("Source config file validation failed ", error.message);
```

File: index.js [hunk 4]
```diff
@@ -2,58 +2,101 @@ const core = require("@actions/core");
...
 const path = require("path");
-const { componentYamlSchemaV1D0 } = require("./schemas");
+const {
+  componentYamlSchemaV1D0,
+  endpointYamlSchemaV0D1,
+  componentConfigYamlSchemaV1beta1,
+} = require("./schemas");
+
+const SourceConfigFileTypes = {
+  COMPONENT_YAML: "component.yaml",
```
Comment (JanakaSandaruwan) [inline index.js]: Adding this in seperate file will increase the reusability. Ignore if not valid

Comment (JanakaSandaruwan) [reply index.js]: ex:- https://github.com/choreo-templates/ballerina-toml-parser/blob/master/enums.js

File: schemas.js [hunk 5]
```diff
@@ -205,6 +248,23 @@ const componentYamlSchemaV1D0 = (srcDir) =>
...
+const endpointYamlSchemaV0D1 = (srcDir) =>
+  yup.object().shape({
+    version: yup.string().required().oneOf(ALLOWED_ENDPOINT_YAML_VERSIONS),
+    endpoints: endpointSchemaV0D1(srcDir).required().min(0),
+  });
+
+// componentConfigYamlSchemaV1D0 - Schema for component-config.yaml
+const componentConfigYamlSchemaV1beta1 = (srcDir) =>
+  yup.object().shape({
+    apiVersion: yup.string().required().oneOf(["core.choreo.dev/v1beta1"]),
```
Comment (JanakaSandaruwan) [inline schemas.js]: Shall we move this to const as others

File: schemas.js [hunk 6]
```diff
@@ -146,6 +162,26 @@ const serviceSchema = yup
   .required()
   .basePathRequired();
 
+// endpointSchemaV0D1 - Schema for endpoint definition V0.1
+const endpointSchemaV0D1 = (srcDir) =>
+  yup.array().of(
+    yup.object().shape({
+      name: yup.string().required(),
```
Comment (JanakaSandaruwan) [inline schemas.js:170]: Any string is allowed for name?

File: schemas.js [hunk 7]
```diff
@@ -137,7 +153,7 @@ const serviceSchema = yup
     basePath: yup
       .string()
       .matches(
-        /^\/[a-zA-Z0-9\/-]*$/,
+        /^\/[a-zA-Z0-9\/-_]*$/,
```
Comment (JanakaSandaruwan) [inline schemas.js:157]: Need to update msg as well?

File: schemas.js [hunk 8]
```diff
@@ -107,7 +123,7 @@ yup.addMethod(yup.string, "validateServiceName", function () {
         return (
           thirdPartySvcRefNameRegex.test(value) ||
           new yup.ValidationError(
-            `${testCtx.path} has an invalid service identifier, ` +
+            `${testCtx?.path} has an invalid service identifier, ` +
```
Comment (JanakaSandaruwan) [inline schemas.js]: Why this is optional?

File: action.yml [hunk 9]
```diff
@@ -1,10 +1,14 @@
...
 inputs:
-  source-root-dir-path: 
+  source-root-dir-path:
     description: "path to the root directory of the source code"
     required: true
     default: ""
+  file-type:
+    description: "type of the file to validate"
+    required: true
+    default: ""
```
Comment (JanakaSandaruwan) [inline action.yml:11]: Can't we set default yaml file as we are recomending component.yaml?

File: schemas.js (hunk 6 above)
Comment (binoyPeries) [reply schemas.js:170]: yes in the old schema in string allowed

File: dist/index.js (hunk 2 above)
Comment (binoyPeries) [reply dist/index.js:38269]: This is needed as when this gets appended to the final err msg, this should start in a new line

File: action.yml (hunk 9 above)
Comment (binoyPeries) [reply action.yml:11]: This might be misleading in a scenario where the yaml is no been present is acceptable. ex: ballerina

**Review by JanakaSandaruwan — APPROVED** [review]

**Pull request merged**